src/main/java/org/folio/dataexp/service/InputFileProcessor.java

```diff
@@ -68,6 +69,16 @@ public void readFile(FileDefinition fileDefinition, CommonExportStatistic common
     }
   }
 
+  public String readMarcFile(String dirName) throws IOException {
```

Comment:
S3Client was designed to operate with streams. This allows to avoid extra memory consumption during transformations IS -> arrays -> string. I would propose here to return IS and refactor code to return InputStreamResource in the processRecordDownload-method.